Examples/LambdaFunctions/Sources/LambdaAPI/ProductLambda.swift

```diff
@@ -0,0 +1,106 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftAWSLambdaRuntime open source project
+//
+// Copyright (c) 2020 Apple Inc. and the SwiftAWSLambdaRuntime project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftAWSLambdaRuntime project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+#if canImport(FoundationNetworking)
+import FoundationNetworking
+#endif
+import AWSLambdaRuntime
+import AWSDynamoDB
+import NIO
+import Logging
+import AsyncHTTPClient
+import AWSLambdaEvents
+
+enum Operation: String {
+    case create
+    case read
+    case update
+    case delete
+    case list
+    case unknown
+}
+
+struct EmptyResponse: Codable {
+    
+}
+
+struct ProductLambda: LambdaHandler {
+    
+    //typealias In = APIGateway.SimpleRequest
+    typealias In = APIGateway.V2.Request
+    typealias Out = APIGateway.V2.Response
+    
+    let dbTimeout:Int64 = 30
+    
+    let region: Region
+    let db: AWSDynamoDB.DynamoDB
+    let service: ProductService
+    let tableName: String
+    let operation: Operation
+
+    var httpClient: HTTPClient
+    
+    static func currentRegion() -> Region {
+        
+        if let awsRegion = ProcessInfo.processInfo.environment["AWS_REGION"] {
```

Comment:
there is a helper for that: Lambda.env

Reply:
also may be worth caching to improve perf

Reply:
- 49aedc7
- It's called only once.